src/DoctrineObject.php

```diff
@@ -330,7 +330,7 @@ public function hydrateValue(string $name, $value, ?array $data = null)
      *
      * @template T of object
      */
-    protected function hydrateByValue(array $data, ?object $object): object
+    protected function hydrateByValue(array $data, object $object): object
```

Comment:
I agree that passing null doesn't make sence, but since the method is `protected` and not `private` this would be a BC break. Please instead add an `@deprecated` annotation stating that object should not be null and that method signature will be changed in the next major.

Reply:
Or better, make in private in next major